src/resmom/stage_func.c

```diff
@@ -1681,7 +1680,7 @@ sys_copy(int dir, int rmtflg, char *owner, char *src, struct rqfpair *pair, int
 				if ((rc = WEXITSTATUS(rc)) == 0) {
 					if ((rmtflg != 0) && (dir == STAGE_DIR_IN)) {
 						if ((stat(ag3, &sb) == -1) ||
-							(original == sb.st_mtime))
+							(original == sb.st_ctime))
 							rc = 13;
```

Comment:
@arungrover in case I wasn't clear earlier, this is where using ctime is important.  Because `scp -p` is used, mtime might not change in value, causing this code to report that the scp failed.  By using ctime, we can detect that the file metadata was changed, which I'm pretty sure always happens when the `-p` option is used, and assume as was done previous that the copy succeeded.

Reply:
Thanks for the explanation @toonen , I am a little surprised that you are hitting this issue at all. Ideally, the PTL test you mentioned should ideally create a temp file and it is highly unlikely (but possible) that it will come up with a name of a file that has already been staged.
If we change the mtime to ctime, it is almost always going to pass, isn't it? Because with each scp, it will update ctime of the file when it is copied. So, should we even be doing this check?

Reply:
I added more details about why a file with the same name already exists to the PR description.

As for the need for the check, the comment at line 1653 states that apparently scp can fail to perform the copy and still return success.  The mtime check was there as an attempt to catch this situation.  Unfortunately, `scp -p` resets mtime possibly back to the same as the original value.  Using ctime is more likely to catch whatever issue resulted in scp failing since it will increase in value when mtime is set.  BTW, if scp fails part way through the copy (having actually written some data) and returns success, then checking for a lack of change in mtime or ctime won't catch the issue.

I don't know what the original situation was when this check was added, so I can't say if it is still necessary.  I can say that in my test environment test_staging fails frequently when the server and mom are not collocated and that switching to ctime fixed the failure.